### PR TITLE
Latest block header testing

### PR DESCRIPTION
**Pull request by piotr-iohk**
- [x] Block header e2e test for light and full mode
- [x] Include 406 response in the swagger

### Issue Number

ADP-1812

### CODE REVIEW

File: specifications/api/swagger.yaml [hunk 1]
```diff
@@ -7639,15 +7647,11 @@ paths:
...
         <p align="right">status: <strong>stable</strong></p>
 
         Return the latest block-header available at the chain source
-      responses: 
-        200:
-          description: OK
-          content:
-            application/json:
-              schema: *ApiBlockHeader
+      responses: *responsesGetBlocksLatestHeader
```
Comment (Unisay) [inline specifications/api/swagger.yaml]: I actually advised @paolino not to extract the `x-responsesGetBlocksLatestHeader` as its only used in one place. 
My logic was that less indirection is better for the readability (local reasoning: you don't have to jump around in order to make sense of a piece of markup).

**Review by Unisay — COMMENTED** [review]
I don't have objections other than the comment about response structure.

File: specifications/api/swagger.yaml (hunk 1 above)
Comment (piotr-iohk) [reply specifications/api/swagger.yaml]: ok, fixed.

**Review by Unisay — APPROVED** [review]
Thanks!

**Pull request merged**